test/extensions/tracers/xray/tracer_test.cc

```diff
@@ -73,6 +80,8 @@ TEST_F(XRayTracerTest, SerializeSpanTest) {
   span->setTag("user_agent", expected_user_agent);
   span->setTag("http.status_code", expected_status_code);
   span->setTag("response_size", expected_content_length);
+  span->setTag("peer.address", expected_client_ip);
+  span->setTag("upstream_address", expected_upstream_address);
```

Comment:
Aren't these *high-cardinailty* tags? Shouldn't we make these opt-in or something?

Reply:
Just worried if there is an added cost for these...

Reply:
My understanding is that we should be reporting all the values we could. And the cost can and should be tuned by the sampling ratio.

Reply:
There is also a limit of 50 annotations on a trace: 

https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-annotations

So adding these counts against what a customer may already have. Plus if they are logging upstream and peer addresses themselves we'd be possibly doing double work.

So AFAICT there is no added cost for more annotations, everything is priced at the 50 annotation limit.